extensions/wikia/NjordPrototype/NjordController.class.php

```diff
@@ -0,0 +1,325 @@
+<?php
+
+class NjordController extends WikiaController {
+
+	const HERO_IMAGE_FILENAME = 'wikia-hero-image';
+	const THUMBNAILER_SIZE_SUFIX = '1600px-0';
+
+	const MAINPAGE_PAGE = 'mainpage';
+
+	private static $wikiDataModel = null;
+
+	public function __construct() {
+		parent::__construct();
+
+		$this->initWikiDataModel();
+	}
+
+	private function initWikiDataModel() {
+		if (self::$wikiDataModel === null) {
+			self::$wikiDataModel = new WikiDataModel(Title::newMainPage()->getText());
+			self::$wikiDataModel->getFromProps();
+		}
+	}
+
+	public function getWikiMarkup() {
+		$articleWikiMarkup = '';
+		$articleTitle = $this->getRequest()->getVal( 'articleTitle' );
+		$pageTitleObj = Title::newFromText( $articleTitle );
+		if ( isset($pageTitleObj) && $pageTitleObj->exists() ) {
+			$pageArticleObj = new Article( $pageTitleObj );
+			$articleWikiMarkup = $pageArticleObj->getPage()->getText();
+		}
+		$this->getResponse()->setFormat( 'json' );
+		$this->getResponse()->setVal( 'wikiMarkup', $articleWikiMarkup );
+	}
+
+	public function MainPageModuleSave() {
+		$articleTitle = $this->getRequest()->getVal( 'pageTitle' );
+		$wikiMarkup = $this->getRequest()->getVal( 'wikiMarkup' );
+		$pageTitleObj = Title::newFromText( $articleTitle );
+		$pageArticleObj = new Article( $pageTitleObj );
+
+		$pageArticleObj->doEdit( $wikiMarkup, '' );
+		$pageArticleObj->doPurge();
+		$newHtml = $pageArticleObj->getPage()->getParserOutput( new ParserOptions( null, null ) )->getText();
+
+		$this->getResponse()->setFormat( 'json' );
+		$this->getResponse()->setVal( 'html', $newHtml );
+	}
+
+	public function index() {
+		global $wgTitle, $wgUser, $wgBlankImgUrl;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$wgTitle" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$wgUser" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)